grafana/grafana
# Logs panel: Support details view

**What this PR does / why we need it**:

This PR brings back logs details view in Logs Panel. We used to have `allowDetails` toggle which name was miss-leading cause `allowDetails === true` actually hided the details view. I have decided to keep this toggling props and add the option to hide log details in Logs panel. It defaults to `false`, so the log view is defaultly visible.

To display links, we are using `getFieldLinksForExplore` which I think wee should rename (but not sure how) as it is used outside of Explore, for example in TimeSeries panel to create links for exemplars. 

When we are using internal data source, it opens an Explore with that data source selected and specified query. If external link, it opens external link. 

The detected fields are filtering are not supported and options is not displayed. Filtering is not possible as we don't have data source in visualisation. For detected fields, we would need some UI work on how to show/toggle it. We can think about it if our users would find this useful. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/22271

**How to test this:**:
1. Run log query with logs panel and open detail vie in the same way as it is done in Explore. You should be able to see:
- Details (labels/detected fields)
- Derived fields with or without link, depending on your field. 
- Possibility to show stats
2. Check if everything stayed correct in Explore.
